apps/speed.c

```diff
@@ -2430,14 +2474,25 @@ int speed_main(int argc, char **argv)
             ret = 0;
             goto end;
         }
-        for (testnum = 0; testnum < size_num; testnum++) {
-            if (evp_cipher) {
 
-                names[D_EVP] = OBJ_nid2ln(EVP_CIPHER_nid(evp_cipher));
-                /*
-                 * -O3 -fschedule-insns messes up an optimization here!
-                 * names[D_EVP] somehow becomes NULL
-                 */
+        loopfunc = EVP_Update_loop;
+        if (evp_cipher != NULL) {
+            names[D_EVP] = OBJ_nid2ln(EVP_CIPHER_nid(evp_cipher));
+
+            if (EVP_CIPHER_mode(evp_cipher) == EVP_CIPH_CCM_MODE) {
+                loopfunc = EVP_Update_loop_ccm;
+            } else if (aead && (EVP_CIPHER_flags(evp_cipher) &
```

Comment:
Why not apply this to every cipher having EVP_CIPH_FLAG_AEAD_CIPHER flag ?

Reply:
Yes, I considered this. Trouble is that those who want to profile low-level subroutines would prefer the original behaviour. So there would be additional flag either way. Being conservative in minor release is more appropriate, i.e. it's more appropriate to engage new behaviour with additional flag and not vice versa.

Reply:
Ok, I see.